taxonomy/apis.py

```diff
@@ -0,0 +1,226 @@
+        "required": vocabulary.required,
+        "vocabulary_type": vocabulary.vocabulary_type,
+        "weight": vocabulary.weight,
+        "learning_object_types": learning_object_types,
+    }
+
+
+def create_vocabulary(
+        repository_id, name, description, required, vocabulary_type, weight,
+        learning_object_types):
+    """
+    Create a vocabulary and save it in database
+    :param repository_id: An existing repository id
+    :param name: Name of vocabulary
+    :param description: Description of vocabulary
+    :param required: Is vocabulary required?
+    :param vocabulary_type: Type of vocabulary (either "managed" or
+     "free tagging")
+    :param weight: Weight, as integer
+    :param learning_object_types: List of valid learning object types
+    :return:
+    """
+
+    learning_object_types_models = LearningObjectType.objects.filter(
+        name__in=learning_object_types)
+
+    vocabulary = Vocabulary.objects.create(
+        repository_id=repository_id,
+        name=name,
+        description=description,
+        required=required,
+        vocabulary_type=vocabulary_type,
+        weight=weight,
+        learning_object_types=learning_object_types_models,
+    )
+
+    return _get_vocabulary_from_model(vocabulary)
+
+
+def update_vocabulary(vocabulary_id,
+                      repository_id, name, description, required,
+                      vocabulary_type, weight,
+                      learning_object_types):
+    """
+    Update an existing vocabulary
+    :param vocabulary_id: An existing vocabulary id
+    :param repository_id: An existing repository id
+    :param name: Name of vocabulary
+    :param description: Description of vocabulary
+    :param required: Is vocabulary required?
+    :param vocabulary_type: Type of vocabulary (either "managed" or
+    "free tagging")
+    :param weight: int
+    :param learning_object_types: List of valid learning object types
+    :return:
+    """
+    learning_object_types_models = LearningObjectType.objects.filter(
+        name__in=learning_object_types)
+
+    Vocabulary.objects.update(
+        id=vocabulary_id,
+        repository_id=repository_id,
+        name=name,
+        description=description,
+        required=required,
+        vocabulary_type=vocabulary_type,
+        weight=weight,
+        learning_object_types=learning_object_types_models,
+    )
+
+
+def delete_vocabulary(vocabulary_id):
+    """
+    Delete vocabulary with given id
+    :param vocabulary_id:
+    :return:
+    """
+    Vocabulary.objects.filter(id=vocabulary_id).delete()
+
+
+def get_term(term_id):
+    """
+    Get term given term id
+    :param term_id: Term id
+    :return: Term
+    """
+    return _get_term_from_model(Term.objects.get(id=term_id))
+
+
+def _get_term_from_model(term):
+    """
+    Convert Term model to dictionary
+    :param term: Term
+    :type term: Term
+    :return: Dictionary
+    """
+    return {
```

Comment:
Instead of these wrapper functions, you can just to model to object with `.values()`https://docs.djangoproject.com/en/dev/ref/models/querysets/#values or django.forms.models.model_to_dict